### DISCUSSION TITLE
Trying to compile MotoROS2 to DX200 target

### DISCUSSION THREAD

**Discussion by Julien44270** [Yaskawa-Global/motoros2#111] [Q&A] [answered]
Hi,

I'm following the above [page](https://github.com/Yaskawa-Global/motoros2/tree/dx200_support#configuration).
Here is the result that I obtain.

![image](https://github.com/Yaskawa-Global/motoros2/assets/141640003/e49f772d-8d58-4d8a-bb4c-4bc2f81b1c80)

MpBuilder is well set up : ![image](https://github.com/Yaskawa-Global/motoros2/assets/141640003/32b554c7-db35-4c01-829c-02ec02a8a23d)

If any mistakes could be reported or evolution would be raised to help to understand, I follow.

Julien

**Comment by gavanderhoorn** [comment]
To get it out of the way: DX200 is not yet officially supported. See the top of the [main README](https://github.com/Yaskawa-Global/motoros2/blob/c284c129940908dbfaad6adb964ef37078a731bd/README.md).

We haven't yet merged #49, so the easy way out for us would be to say: "please wait a bit for official DX200 support".

However, I believe @ted-miller has successfully built #49 using mpBuilder.

Could you please clarify what exactly you have done, which files you've downloaded, which branches you've checked out, how you've setup everything, etc?

**Reply by Julien44270** [reply]
I forked [this one](https://github.com/Yaskawa-Global/motoros2/tree/dx200_support) that I put on a dedicated workspace on my computer for working.
I put the corresponding [library](https://github.com/Yaskawa-Global/motoros2/releases/download/0.1.0/1691071948_libmicroros_humble-20221102-dbg_dx200.zip), following the described structure.
A SDK DN key is well connected to the computer that I used to realize the compilation.

I will meet the above information to go on with, and I'll come back with other data.

**Reply by ted-miller** [reply]
I'm pretty sure this error is caused by the version of MotoPlus SDK. The structure of this project required updates to the SDK itself. MotoROS2 [requires v1.5.1](https://github.com/Yaskawa-Global/motoros2/blob/main/doc/build_from_source.md#requirements) or greater of MotoPlus SDK. 

ccs@yaskawa.eu should be able to provide you an update that includes the license libraries for your region. If there's any issues, let us know.

**Reply by Julien44270** [reply]
I've just upgraded the DX200 controller (from 2.xxx to 3.xxx).
I'm blocking about transfering the yaml config file necessary for the deployment.
The support indicates that a lack of memory or the size of the file (too big) could be an issue...
Would you know if a memory extension (like SRAM) could solve it?

**Reply by gavanderhoorn** [reply, marked as answer]
If you're referring to the steps in the documentation where it asks you to save the `.yaml` on the SRAM drive as a *user defined file*: that's not used on DX200.

Place the `.yaml` on a USB stick (FAT32 formatted) and insert it into `CN106`. Then reboot.

(I'd like to reiterate we don't officially yet support DX200. This is one of those areas where the documentation still needs to be updated/clarified more)

**Comment by gavanderhoorn** [comment]
@Julien44270: just wanted to let you know we've just released `0.1.1`, which includes support for DX200 controllers. See the [releases](https://github.com/Yaskawa-Global/motoros2/releases/tag/0.1.1).

You also don't need to build it from source any more, as we've included DX200 binaries in this release.
